File: src/money/MoneyTest.java
package money;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

class MoneyTest {

	@Test
	public void testMulti() {
		Money five = Money.dollar(5);
		assertEquals(new Dollar(10), five.times(2));
		assertEquals(new Dollar(15), five.times(3));
	}
	
	@Test
	public void testEqual() {
		assertTrue(new Dollar(5).equals(new Dollar(5)));
		assertFalse(new Dollar(5).equals(new Dollar(6)));
		assertTrue(new Franc(5).equals(new Franc(5)));
		assertFalse(new Franc(5).equals(new Franc(6)));
		assertFalse(new Franc(5).equals(new Dollar(5)));
	}
	
	@Test
	public void testFrancMulti() {
		Franc five = new Franc(5);
		assertEquals(new Franc(10), five.times(2));
		assertEquals(new Franc(15), five.times(3));
	}

}

File: ruby/money_test.rb
require 'minitest/autorun'
require_relative 'money'
require_relative 'dollar'
require_relative 'franc'

class MoneyTest < Minitest::Test
  def test_multiplication
    five = Money.dollar(5)
    assert_equal Money.dollar(10), five.times(2)
    assert_equal Money.dollar(15), five.times(3)
  end

  def test_equality
    assert_equal(true, Money.dollar(5) == (Money.dollar(5)))
    assert_equal(false, Money.dollar(5) == (Money.dollar(6)))
    assert_equal(false, Franc.new(5) == (Franc.new(6)))
    assert_equal(false, Franc.new(5) == (Franc.new(6)))
    assert_equal(false, Franc.new(5) == (Money.dollar(5)))
  end

  def test_franc_multiplication
    five = Franc.new(5)
    assert_equal Franc.new(10), five.times(2)
    assert_equal Franc.new(15), five.times(3)
  end
end
